CMakeLists.txt

```diff
@@ -37,6 +37,11 @@ else()
 		add_compile_options(-fdiagnostics-color)
 	endif()
 
+	check_cxx_compiler_flag(-Wno-maybe-uninitialized flag_no_maybe_uninit)
+	if (flag_no_maybe_uninit)
+	add_compile_options(-Wno-maybe-uninitialized)
```

Comment:
Can you use `target_compile_options` to ensure this applies to the GTest library only. It is better to keep the warning for Ninja and fix the errors in the source code directly instead.

Also please add a comment that explains why this section exists in the first place, e.g. it can be removed in the future, for example once googlegtest is upgraded to a newest release that doesn't have this problem. Ideally a version check here would be a good idea too, but I don't know if this is possible here.

Reply:
hh, thanks for your advice, works before looks aggresive a little bit.